pages/index.tsx

```diff
@@ -1,7 +1,124 @@
-import type { NextPage } from 'next'
+import type { GetStaticProps, NextPage } from 'next'
+import { useState } from 'react'
+import { v4 } from 'uuid'
+import { useTranslations } from 'next-intl'
 
-const Home: NextPage = () => {
-  return <div>TODO</div>
+import { Question, renderQuestion } from '../components/util/questionRenderer'
+import styles from '../styles/base.module.css'
+import config from '../config.yaml'
+
+const sessionUuid = v4()
+
+type Props = {
+  children?: React.ReactNode
+  questions?: Question[]
+}
+
+const Home: NextPage = ({ questions }: Props) => {
+  const [activeQuestion, setActiveQuestion] = useState(0)
+  const t = useTranslations()
+
+  const updateHandler = (update: unknown) => {
+    console.log(
+      `update! session ${sessionUuid}. activeQuestion ${activeQuestion}. value ${update}`
+    )
+    fetch(`/api/recordUpdate`, {
+      body: JSON.stringify({
+        question: activeQuestion,
+        sessionUuid,
+        value: update
+      }),
+      method: 'POST'
+    })
+  }
+
+  if (!questions || questions.length === 0) {
+    return <div>No questions defined</div>
+  }
+
+  const surveyOver = activeQuestion >= questions.length
+
+  const prevQuestion = () => setActiveQuestion(activeQuestion - 1)
+  const nextQuestion = () => setActiveQuestion(activeQuestion + 1)
+
+  return (
+    <main className={styles.wrapper}>
+      {/* dynamically imported content */}
+      {questions.map((question, index) => {
+        if (question.i18n) {
+          question.title = t(`question${index}.title`)
+        }
+        return (
+          <form
+            className={styles.question}
+            key={index}
+            onSubmit={nextQuestion}
+            style={{
+              display: index === activeQuestion ? 'inherit' : 'none'
+            }}
+          >
+            {renderQuestion(question, (update: unknown) =>
+              activeQuestion === index
+                ? updateHandler(update)
+                : console.log(
+                    'Swallowing invalid callback update provided by inactive question'
+                  )
+            )}
+          </form>
+        )
+      })}
+
+      {surveyOver && (
+        <div className={styles.question}>
+          <h1>{t('index.completion')}</h1>
+          <h2>{t('index.completionSecondary')}</h2>
+        </div>
+      )}
+
+      <section className={styles.buttons}>
```

Comment:
Submitting a form via button causes a page refresh -- one could override this by using `preventDefault`, but then that makes the buttons getting re-rendered on question change more confusing. 

I think if we did this all questions should be in the form unlike now where each question is its own form

Reply:
It would be better if all questions were in one form imo.

Reply:
I've thought about this -- it would be possible, but each question needs to submit on keystroke/click so it would be a form without an onsubmit handler. Also, the next and previous buttons would now have no impact on a screen reader user.